src/App.js

```diff
@@ -1,15 +1,45 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
-
 function App() {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+      new Promise((resolve, reject) =>
+        setTimeout(() => {
+          resolve({ data: { todoList: JSON.parse(localStorage.getItem('savedTodoList')) || [] }}
```

Comment:
Careful here, `[]` is not JSON. If `savedTodoList` was empty you'll get an error. To account for the scenario of an empty or null `savedTodoList`, you would want to use the spread operator on the result.data.todoList inside of a new array.

`setTodoList([...result.data.todoList]);`
if todoList was empty, you'll end up with an empty array which is the desired outcome.

Reply:
Here's what happens if you do run `JSON.parse` in the console...
<img width="432" alt="202301_0516PM-Brave Browser" src="https://user-images.githubusercontent.com/10133316/212570085-f06843f7-a72f-443b-953c-91ebf9c279de.png">